### PR TITLE
Add the shared infrastructure for testing CEReactions IDL attribute

### PR DESCRIPTION
**Pull request by rniwa**
and test relevant attributes on ChildNode, Document, Element, Node, and ParentNode interfaces.

### CODE REVIEW

**Comment by rniwa** [comment]
@hayatoito @annevk @domenic @dominiccooney @Ms2ger @travisleithead 

I'm open to changing the way things are setup here.  I just didn't want to continue writing the rest of the tests before getting some feedback on the common testing infrastructure.

File: custom-elements/reactions/resources/reactions.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+function defineNewCustomElement(observedAttributes) {
+    let log = [];
+    let name = 'custom-element-' + testNumber++;
+
+    class CustomElement extends HTMLElement {
+        constructor() {
+            super();
+            log.push({type: 'constructed', element: this});
+        }
+        attributeChangedCallback(name, oldValue, newValue) { log.push({type: 'attributeChanged', element: this, name: name, oldValue: oldValue, newValue: newValue}); }
```
Comment (domenic) [inline custom-elements/reactions/resources/reactions.js]: I guess it's probably worth adding namespace here too, even if it's a very unloved argument.

**Comment by domenic** [comment]
This looks like a really nice setup to me. I'll try to get @dominiccooney to comment since he's been coordinating most of our tests.

**Comment by dominiccooney** [comment]
Our tests tend to use iframes so they don't have to gensym names. For consistent failures it might be nice to not gensym the names because I don't think the precise order of things in an async test are defined.

**Comment by rniwa** [comment]
@dominiccooney : what are you referring to by "gensym names"?

**Comment by dominiccooney** [comment]
The way it creates a fresh name by doing 'stringliteral' + counter++.

**Comment by dominiccooney** [comment]
I think this is good as-is, and we should upstream Google's tests, rationalize our test_with_window thing and then work upstream to make the tests use the helpers consistently.

**Pull request merged**